it/endpoints/GetReturnsControllerISpec.scala

```diff
@@ -0,0 +1,344 @@
+            countryCode = "GB"
+          ),
+          totalCGTLiability = 0,
+          charges = None
+        )
+
+        DownstreamStub.onSuccess(
+          method = DownstreamStub.GET,
+          uri = listDownstreamUri,
+          queryParams = listDownstreamQueryParams,
+          status = OK,
+          body = Json.obj("returnList" -> JsArray(Seq(Json.toJson(sampleDesReturn))))
+        )
+
+        taxYearQueryDates.foreach(params =>
+          DownstreamStub.onError(
+            method = DownstreamStub.GET,
+            uri = getFinancialDataDownstreamUri,
+            queryParams = params,
+            errorStatus = NOT_FOUND,
+            errorBody = notFoundFinancialDataErrorBody.toString()
+          )
+        )
+
+        val amendReturnsRepository: AmendReturnsRepository = app.injector.instanceOf[DefaultAmendReturnsRepository]
+
+        val sampleSubmitReturnRequest: SubmitReturnRequest = sample[SubmitReturnRequest]
+
+        val sampleSubmitReturnRequestWithCgtId: SubmitReturnRequest = sampleSubmitReturnRequest.copy(
+          subscribedDetails = sampleSubmitReturnRequest.subscribedDetails.copy(
+            cgtReference = CgtReference(cgtRef)
+          )
+        )
+
+        val sampleAmendReturnData: AmendReturnData = sample[AmendReturnData]
+
+        val sampleAmendReturnDataWithReturnId: AmendReturnData = sampleAmendReturnData.copy(
+          originalReturn = sampleAmendReturnData.originalReturn.copy(
+            summary = sampleAmendReturnData.originalReturn.summary.copy(
+              submissionId = "someSubmissionId"
+            )
+          )
+        )
+
+        val aa: EitherT[Future, models.Error, Unit] = amendReturnsRepository.save(
+          submitReturnRequest = sampleSubmitReturnRequestWithCgtId.copy(
+            amendReturnData = Some(sampleAmendReturnDataWithReturnId)
+          )
+        )
+
+        val resultJson: JsValue = Json.obj(
+          "returns" -> JsArray(
+            Seq(
+              Json.obj(
+                "submissionId"           -> sampleDesReturn.submissionId,
+                "submissionDate"         -> sampleDesReturn.submissionDate,
+                "completionDate"         -> sampleDesReturn.completionDate,
+                "taxYear"                -> sampleDesReturn.taxYear,
+                "mainReturnChargeAmount" -> 0,
+                "propertyAddress"        -> Json.obj(
+                  "UkAddress" -> Json.obj(
+                    "line1"    -> sampleDesReturn.propertyAddress.addressLine1,
+                    "line2"    -> sampleDesReturn.propertyAddress.addressLine2,
+                    "town"     -> sampleDesReturn.propertyAddress.addressLine3,
+                    "postcode" -> sampleDesReturn.propertyAddress.postalCode
+                  )
+                ),
+                "charges"                -> JsArray.empty,
+                "isRecentlyAmended"      -> true,
+                "expired"                -> false
```

Comment:
⛏️ Is there no way to construct this with Writes?

Reply:
There is I was just being lazy